### PR TITLE
admin csv QA

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
QA items for Admin CSV download:

1. Set MIMEtype when generating a presigned S3 URL. 
2. Format specific CSV column values 

## WHY
1. Otherwise, Chrome will change the downloaded file's extension from `csv -> txt`
2. So that they look better to the user. 

## HOW
1. Fog gem config 
2. Add a switch statement in the CSV generator class, which applies various transforms. 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Admin-Data-Export-QA-November-6-f1bfffdb6c7d45a9b2705c6883456e6d

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? |
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)

### CODE REVIEW

File: services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb [hunk 1]
```diff
@@ -6,43 +6,66 @@ module Adapters
...
           end
         end
       end
 
-      def self.validate_input!(bigquery_result)
+      def self.human_displayable_csv_headers(sym_columns)
+        sym_columns.map{|col| ORDERED_COLUMNS[col] }
+      end
+
+      def self.format_cell(sym_column, value)
```
Comment (brendanshean) [inline services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb:43]: I tend to prefer `when` ... `then` one-liners if the instruction is short:

```
def format_cell(sym_column, value)
  case sym_column
    when :completed_at then value&.strftime('%F')
    when :timespent then (value / 60).to_i
    when :score then value.respond_to?(:*) ? "#{(value * 100).round}%": ''
    else value.to_s
  end
end
```

File: services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb [hunk 2]
```diff
@@ -6,43 +6,66 @@ module Adapters
...
+        case sym_column
+        when :completed_at
+          value&.strftime("%F")
+        when :timespent
+          (value / 60).to_i
+        when :score
+          value.respond_to?(:*) ? "#{(value*100).round(0)}%" : ''
```
Comment (brendanshean) [inline services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb]: Unless we're expecting some other type, what about `value.is_a?(Numeric)`

**Review by brendanshean — APPROVED** [review]
Some minor stylistic comments but looks good.

File: services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb [hunk 3]
```diff
@@ -6,43 +6,66 @@ module Adapters
...
+      def self.human_displayable_csv_headers(sym_columns)
+        sym_columns.map{|col| ORDERED_COLUMNS[col] }
+      end
+
+      def self.format_cell(sym_column, value)
+        case sym_column
+        when :completed_at
+          value&.strftime("%F")
+        when :timespent
+          (value / 60).to_i
```
Comment (anathomical) [inline services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb]: I'm pretty sure that `timespent` is an Integer, so this will do integer division, meaning we shouldn't need to cast the `to_i` at the end.  Unless Ruby is interpreting the BigQuery data differently than it does Postgres.

File: services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb [hunk 4]
```diff
@@ -6,43 +6,66 @@ module Adapters
...
+        when :score
+          value.respond_to?(:*) ? "#{(value*100).round(0)}%" : ''
+        else
+          value
+        end
+      end
+
+      def self.row_contains_requested_columns?(row, requested_columns)
+        (row.keys & requested_columns).length == requested_columns.length
+      end
```
Comment (anathomical) [inline services/QuillLMS/app/queries/adapters/csv/admin_premium_data_export.rb:58]: If, somehow, `requested_columns` ended up having duplicate key names, this comparison will fail as `&` will collapse duplicates.  My bet is that we have no reason to believe that this will ever happen, so no need to make a change here, but I wanted to flag it to make sure we'd at least considered the possibility.

File: services/QuillLMS/spec/queries/adapters/csv/admin_premium_data_export_spec.rb [hunk 5]
```diff
@@ -5,6 +5,14 @@
 describe Adapters::Csv::AdminPremiumDataExport do
   subject { described_class }
 
+  describe '#format_cell' do
+    it { expect(described_class.format_cell(:completed_at, DateTime.new(2020,1,1))).to eq('2020-01-01')}
+    it { expect(described_class.format_cell(:activity_pack, 'foo')).to eq('foo')}
+    it { expect(described_class.format_cell(:timespent, 61)).to eq(1)}
+    it { expect(described_class.format_cell(:score, 0.66777)).to eq('67%')}
+    it { expect(described_class.format_cell(:not_a_special_case, 1.5)).to eq(1.5)}
```
Comment (anathomical) [inline services/QuillLMS/spec/queries/adapters/csv/admin_premium_data_export_spec.rb:13]: I always appreciate when arbitrary values like this help me understand what the test case is.  Nice!

**Review by anathomical — APPROVED** [review]
I flagged a possible redundant `to_i` call, and an edge case that I don't think we need to worry about, but felt was worth considering.  But overall this looks good to me.
